src/coreclr/tools/r2rdump/TextDumper.cs

```diff
@@ -561,13 +561,55 @@ internal override void DumpFixupStats()
                     Count = group.Count()
                 }).OrderByDescending(x => x.Count);
 
-            Console.WriteLine($"                      Fixup | Count");
+            /* Runtime Repo GH Issue #49249:
+             *
+             * In order to format the fixup counts results table, we need to
+             * know beforehand the size of each column. */
+
+            int longestFixupKind = 0;
+            int sortedFixupCountsTotal = sortedFixupCounts.Sum(x => x.Count);
+
+            /* First, we look at all the Fixup Kinds that will be printed. We
```

Comment:
are the lengths very dynamic, i believe we have a well known list of these fixup kinds?

Reply:
I think we do have a list somewhere (I haven't seen it), but I also made it dynamic because depending on the image dumped, we can get different fields. A good example of this is comparing my output to Tomas' in the original issue. Same as the count values. I was also wondering about new fields that might be added in the future when I worked on this.

Reply:
This is the list of fixups:

https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/coreclr/inc/readytorun.h#L159

I however think that dynamic size determination that Ivan implemented is cleaner because otherwise we would need to hardcode some value that would be prone to becoming stale as we potentially add new fixups exactly as Ivan says.